Shared/Samples/Geocode offline/GeocodeOfflineView.Model.swift

```diff
@@ -0,0 +1,206 @@
+        @Published var isShowingErrorAlert = false
+        
+        /// The error shown in the error alert.
+        @Published private(set) var error: Error? {
+            didSet { isShowingErrorAlert = error != nil }
+        }
+        
+        init() {
+            graphicsOverlay.addGraphic(markerGraphic)
+        }
+        
+        // MARK: Methods
+        
+        /// Geocodes a given address and adds a marker with the corresponding address at the result's location.
+        /// - Parameter address: The given text address to geocode.
+        func geocodeSearch(address: String) async {
+            guard let locatorTask else { return }
+            
+            // Create geocode parameters.
+            let geocodeParameters = GeocodeParameters()
+            geocodeParameters.addResultAttributeName("Match_addr")
+            geocodeParameters.minScore = 75
+            
+            do {
+                // Perform geocode using the locator task with the text address and parameters.
+                let geocodeResults = try await locatorTask.geocode(
+                    forSearchText: address,
+                    using: geocodeParameters
+                )
+                
+                if let result = geocodeResults.first,
+                   let displayLocation = result.displayLocation {
+                    // If a result is found, place a marker at the result's location.
+                    let resultText = result.attributes["Match_addr"] as? String ?? ""
+                    updateMarker(to: displayLocation, withText: resultText)
+                    
+                    // Zoom to the extent of the result's location.
+                    viewpoint = Viewpoint(boundingGeometry: displayLocation.extent)
+                } else {
+                    // If no result was found, inform the user with an alert.
+                    throw "No results found"
+                }
+            } catch {
+                self.error = error
+            }
+        }
+        
+        /// Reverse geocodes a given location and adds a marker with the corresponding address at the result's location.
+        /// - Parameter mapPoint: The point on the map to reverse geocode.
+        func reverseGeocode(mapPoint: Point) async {
+            guard let locatorTask else { return }
+            
+            //  Normalized the map point.
+            guard let normalizedPoint = GeometryEngine.normalizeCentralMeridian(
+                of: mapPoint
+            ) as? Point else { return }
+            
+            // Create reverse geocode parameters.
+            let reverseGeocodeParameters = ReverseGeocodeParameters()
+            reverseGeocodeParameters.addResultAttributeName("*")
+            reverseGeocodeParameters.maxResults = 1
+            
+            // Perform reverse geocode using the locator task with the point and parameters.
+            let geocodeResults = try? await locatorTask.reverseGeocode(
```

Comment:
It may be good to keep the error handling consistent for the geocode/reverseGeocode methods. In the method above it throws whereas here the error is omitted.

I'm fine with omitting both as it is unlikely to have null inputs in our case and there isn't network connection involved for offline.

Reply:
Sounds good. The reason I omitted this one it is and occasionally throw a cancellation error when dragging.